src/app/modules/contract-printables/sales-contract/sales-contract.component.html

```diff
@@ -35,9 +35,11 @@ <h1><b>CONTRACT</b></h1>
       <div class='column-size-3' *ngIf="contractForm.date">
         <b><span [printableField]="'id'">{{ contractForm.id || 0 }}</span>-{{ contractForm.date.getFullYear() % 100 }}</b><br>
         <!-- {{ contractForm.price | getPricePerUnit:"bu":product }} -->
-        {{ (contractForm.price | pricePerUnit:"bu":contractForm.quantity:contractForm.price.amount:contractForm.price.unit:contractForm.quantity.amount:contractForm.quantity.defaultUnits) - contractForm.market_price | number:"0.5-5" }}<br>
-        <span [printableField]="'market_price'">{{ contractForm.market_price | number:"0.5-5" }}</span><br>
-        <span [printableField]="'price.amount'">{{ contractForm.price | pricePerUnit:"bu":contractForm.quantity:contractForm.price.amount:contractForm.price.unit:contractForm.quantity.amount:contractForm.quantity.defaultUnits | number:"0.5-5" }}</span><br>
+        <span [printableField]="'base'">{{ contractForm.base | number:"0.5-5" }}</span><br>
```

Comment:
Removing previous line makes the older contracts not show the base, since the field was not used before.

Fix this by setting the base in the Contract constructor if the market_price and price fields are set

Reply:
Followed your suggestion and updated the contract class to set the base in that case.